<?xml version="1.0" ?>
<launch>
  <param name="use_sim_time" value="true"/>
  <arg name="path" default="/home/$(optenv USER gsl)/datasets/omni_sim/omni_simulated.bag"/>
  <arg name="debug" default="false"/>
  <arg name="publish" default="true"/>
  <arg name="rate" default="1.0"/>
  <arg name="particles" default="250"/>
  <arg name="performer_path" default="/home/$(optenv USER gsl)/datasets/pfuclt/"/>
  <arg name="performer_file" default="experimentX-y_Nrobots_datasetD_runR.bag"/>
  <node pkg="rosbag" type="play" name="player" required="true" output="screen" args="--quiet --clock --rate=$(arg rate) $(arg path)"/>
  <node name="performer" pkg="pfuclt_omni_dataset" type="pfuclt_omni_dataset" output="screen" args="--debug $(arg debug) --publish $(arg publish)">
    <param name="percentage_to_keep" value="50"/>
    <param name="velocity_estimator_stack_size" value="10"/>
    <param name="predict_model_stddev" value="30"/>
    <param name="particles" value="$(arg particles)"/>
    <param name="OMNI1_alpha" value="0.015, 0.1, 0.5, 0.001"/>
    <remap to="/gtData" from="/gtData_4robotExp"/>
    <rosparam param="PLAYING_ROBOTS">[1]</rosparam>
    <param name="MAX_ROBOTS" value="1"/>
    <param name="ROB_HT" value="0.81"/>
    <param name="MY_ID" value="1"/>
    <param name="NUM_TARGETS" value="1"/>
    <param name="LANDMARK_COV/K1" value="0.02"/>
    <param name="LANDMARK_COV/K2" value="0.005"/>
    <param name="LANDMARK_COV/K3" value="0.002"/>
    <param name="LANDMARK_COV/K4" value="0.005"/>
    <param name="LANDMARK_COV/K5" value="0.005"/>
    <param name="NUM_LANDMARKS" value="10"/>
    <param name="LANDMARKS_CONFIG" value="$(find pfuclt_omni_dataset)/config/landmarks.csv"/>
    <param name="USE_CUSTOM_VALUES" value="false"/>
    <rosparam param="POS_INIT">[4.92127393067666, -2.1573843429859787, -0.674671993798972]</rosparam>
    <rosparam param="CUSTOM_PARTICLE_INIT">[4.901273930676661,4.94127393067666,-2.1773843429859787,-2.1373843429859787,-0.694671993798972,-0.654671993798972,5.761477374919562,5.801477374919561,-2.04470759833967,-2.00470759833967,1.5046100813899537,1.5446100813899537]</rosparam>
  </node>
  <node pkg="rosbag" name="record_performance" type="record" args="--all --output-name=$(arg performer_path)/$(arg performer_file)"/>
</launch>
